woof-code/rootfs-skeleton/etc/rc.d/rc.shutdown

```diff
@@ -130,10 +136,26 @@ for i in /sys/class/net/* ; do
 	fi
 done
 
-#terminate dbus daemon
-killall dbus-daemon 2>/dev/null
+#=======================================================================
 
-killall udevd > /dev/null 2>&1
+# terminate stuff
+# dbus daemon
+if [ -e /var/run/dbus/dbus.pid ]; then
+	DPID=$(cat /var/run/dbus/dbus.pid)
+	kill $DPID >/dev/null 2>&1
+	rm -f /var/run/dbus/dbus.pid
+fi
+killall dbus-daemon >/dev/null 2>&1
+# udev
+udevadm control --exit
+killall udevd >/dev/null 2>&1
+# pulse
+pidof pulseaudio >/dev/null 2>&1 && killall pulseaudio >/dev/null 2>&1
```

Comment:
This `pidof` finds the PID of the process called `pulseaudio`, then `killall` does exactly the same - if errors are silenced anyway, `killall pulseaudio >/dev/null 2>&1` is enough

Reply:
Ok, so no need to check if it's running.